iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/stack.inl

```diff
@@ -30,10 +30,10 @@ inline cxx::optional<T> stack<T, Capacity>::pop() noexcept
         return cxx::nullopt;
     }
 
-    // low level memory management with access to the topmost element on the untyped buffer;
-    // type safety is ensured by the template parameter
+    // AXIVION Next Construct AutosarC++19_03-A5.2.4 : low level memory management with access to the topmost element on
+    // the untyped buffer; reinterpret_cast is safe since the size and the alignment of each array element is guaranteed
     // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
-    return *reinterpret_cast<T*>(m_data[--m_size]);
+    return *reinterpret_cast<T*>(&m_data[--m_size]);
```

Comment:
As discussed, I still wonder why it worked before. I think the errors where compensating each other but I would have expected illegal memory accesses.